dask/array/tests/test_optimization.py

```diff
@@ -117,13 +116,6 @@ def test_hard_fuse_slice_cases():
     assert optimize_slices(dsk) == {'x': (getarray, 'x', (None, 5))}
 
 
-def test_dont_fuse_different_slices():
```

Comment:
Wouldn't it be better to synthesize the desired dask graph instead of relying on rechunk()'s implementation details?
(note I'm not sure what exactly is being checked here, though the test name gives a vague hint)